pkg/i2gw/providers/istio/types.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package istio
+
+import (
+	istioapi "istio.io/api/networking/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	APIVersion         = "networking.istio.io/v1beta1"
+	GatewayKind        = "Gateway"
+	VirtualServiceKind = "VirtualService"
+)
+
+type gateway struct {
```

Comment:
@howardjohn would this be an ok representation of the object to parse into from the `unstructured.Unstructured`? Or is there something similar in the istio API (adapted Istio resources for K8S)? 

I didn't find any, asking just to double check.

Reply:
istio/client-go repo already has these that can be used

Reply:
Cool, exactly what's needed. Thanks!